pkg/filters/redirector/redirector.go

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright (c) 2017, MegaEase
+ * All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package redirector
+
+import (
+	"regexp"
+	"strings"
+
+	"github.com/megaease/easegress/pkg/context"
+	"github.com/megaease/easegress/pkg/filters"
+	"github.com/megaease/easegress/pkg/logger"
+	"github.com/megaease/easegress/pkg/protocols/httpprot"
+	"github.com/megaease/easegress/pkg/util/stringtool"
+)
+
+const (
+	// Kind is the kind of Redirector.
+	Kind = "Redirector"
+
+	resultMismatch = "mismatch"
+)
+
+const (
+	matchPartFull = "full"
+	matchPartURI  = "uri"
+	matchPartPath = "path"
+)
+
+var statusCodeMap = map[int]string{
+	301: "Moved Permanently",
+	302: "Found",
+	303: "See Other",
+	304: "Not Modified",
+	307: "Temporary Redirect",
+	308: "Permanent Redirect",
+}
+
+var kind = &filters.Kind{
+	Name:        Kind,
+	Description: "Redirector redirect HTTP requests.",
+	Results:     []string{resultMismatch},
+	DefaultSpec: func() filters.Spec {
+		return &Spec{}
+	},
+	CreateInstance: func(spec filters.Spec) filters.Filter {
+		return &Redirector{spec: spec.(*Spec)}
+	},
+}
+
+func init() {
+	filters.Register(kind)
+}
+
+type (
+	// Redirector is filter to redirect HTTP requests.
+	Redirector struct {
+		spec *Spec
+		re   *regexp.Regexp
+	}
+
+	// Spec describes the Redirector.
+	Spec struct {
+		filters.BaseSpec `json:",inline"`
+
+		Match       string `json:"match" jsonschema:"required"`
+		MatchPart   string `json:"matchPart" jsonschema:"required,enum=uri|full|path"` // default uri
```

Comment:
propose to improve the name `matchPart`, as `match` is a regular expression, it seems we don't need to distinguish `uri` and `path`.